installer/api.py

```diff
@@ -237,22 +239,20 @@ def __getLogTail() -> Optional[str]:
 	# Return content if it went right
 	return output if retCode == 0 else None
 
-def __getArchitectureName() -> str:
-	"""
-	### This function returns the name of the system's architecture name.
 
-	#### Returns:
-	- (str): Architecture name.
-	"""
-	# Initializing to unknown value
-	archName = UNKNOWN_VALUE
 
-	# Obtaining architecture name
-	retCode, architecture = runJob('cat /sys/devices/cpu/caps/pmu_name')
+def __getCPUFlags(lscpu_output) -> str:
+    lines = lscpu_output.splitlines()
+    flags = []
+    capture = False
 
-	# If command worked and returned info, extract it
-	if retCode == 0 and architecture:
-		archName = architecture
-	
-	# Returing architecture name
-	return archName
+    for line in lines:
+        if "Flags:" in line:
+            capture = True
+            flags.append(line.split("Flags:")[1].strip())
+        elif capture:
+            if line.startswith(" "):
+                flags.append(line.strip())
```

Comment:
i guess once you are within the flags part of the text, you can always `strip()` (if the line does not have starting/trailing spaces it won't change), then `split(" ")` to get an element for every flag, and finally, `flags.extend()`, which adds all the elements of the splitted list without nesting them inside the flags list.

However, this is all theoretical and have not tested it, i will be impementing it in the (new-new) installer today, so check the issue i mentioned in another comment to check when it is closed, go to the PR that closed it, and check there my implementation as possible reference/inspiration.

Reply:
Update to this: I made this function
```py
def __get_compilation_flags() -> List[str]:
	"""
	### This function obtains the list of compilation flags supported by the CPU.

	#### Returns:
	- (list(str)): List of flags supported by the CPU.  
	"""
	flags_header = "Flags:"
	ret_code, cpu_info = shell_handler.run_shell_command(f'lscpu | grep "{flags_header}"')
	if ret_code:
		return []
	cpu_info = cpu_info.replace(flags_header, "").strip()
	return cpu_info.split(" ")
```
Which works fine, as the multiple lines printed by the `lscpu` command in the flags section is only due to the screen size, but they are actually in the same line string-ly speaking (there is no `\n` between those lines). That means that command `lscpu | grep "Flags:"` will filter that output to return only that line. Then, you remove the "Flags:" part, strip, and split, which gives you the list.